java/query/query-lambda/src/test/java/sleeper/query/lambda/WarmQueryExecutorLambdaIT.java

```diff
@@ -0,0 +1,282 @@
+import static sleeper.query.runner.output.NoResultsOutput.NO_RESULTS_OUTPUT;
+
+@Testcontainers
+public class WarmQueryExecutorLambdaIT {
+    @Container
+    public static LocalStackContainer localStackContainer = new LocalStackContainer(
+            DockerImageName.parse(CommonTestConstants.LOCALSTACK_DOCKER_IMAGE))
+            .withServices(LocalStackContainer.Service.S3, LocalStackContainer.Service.SQS,
+                    LocalStackContainer.Service.DYNAMODB);
+
+    private final AmazonS3 s3Client = buildAwsV1Client(localStackContainer, LocalStackContainer.Service.S3,
+            AmazonS3ClientBuilder.standard());
+    private final AmazonDynamoDB dynamoClient = buildAwsV1Client(localStackContainer,
+            LocalStackContainer.Service.DYNAMODB, AmazonDynamoDBClientBuilder.standard());
+    private final AmazonSQS sqsClient = buildAwsV1Client(localStackContainer, LocalStackContainer.Service.SQS,
+            AmazonSQSClientBuilder.standard());
+    private final Configuration configuration = getHadoopConfiguration(localStackContainer);
+    private QuerySerDe querySerDe;
+
+    @TempDir
+    public Path tempDir;
+    private InstanceProperties instanceProperties;
+    protected static WarmQueryExecutorLambda lambda;
+
+    @BeforeEach
+    void setUp() throws IOException, ObjectFactoryException {
+        String dataDir = createTempDirectory(tempDir, null).toString();
+        createInstanceProperties(dataDir);
+        lambda = new WarmQueryExecutorLambda(s3Client, sqsClient, dynamoClient, instanceProperties.get(CONFIG_BUCKET));
+    }
+
+    @AfterEach
+    void tearDown() {
+        s3Client.shutdown();
+        dynamoClient.shutdown();
+        sqsClient.shutdown();
+    }
+
+    @Test
+    public void shouldCreateAQueryWithKeyTypeOfString() throws Exception {
+        // Given
+        Schema schema = getStringKeySchema();
+        TableProperties tableProperties = createTestTableProperties(instanceProperties, schema);
+        createTable(tableProperties);
+        querySerDe = new QuerySerDe(schema);
+
+        // When
+        lambda.handleRequest(new ScheduledEvent(), null);
+
+        // Then
+        ReceiveMessageResult result = sqsClient.receiveMessage(new ReceiveMessageRequest(instanceProperties.get(QUERY_QUEUE_URL)));
+        assertThat(result.getMessages()).hasSize(1);
+
+        Query query = querySerDe.fromJson(result.getMessages().get(0).getBody());
+        assertThat(query.getTableName().equals(tableProperties.get(TABLE_NAME)));
+        assertThat(query.getResultsPublisherConfig().get(ResultsOutputConstants.DESTINATION).equals(NO_RESULTS_OUTPUT));
+        query.getRegions().forEach(region -> {
+            Range range = region.getRange("key");
+            assertThat(range.getField().getType()).isInstanceOf(StringType.class);
+            assertThat(range.getMin()).isEqualTo("a");
+            assertThat(range.getMax()).isEqualTo("a");
+        });
```

Comment:
If you build the query you expect and assert on exact equality with the query object, the assertion will be stricter and the output will be more useful when it fails. There's a lot that isn't asserted here. This is the same for all the tests.